lib/npg_qc/autoqc/checks/pulldown_metrics.pm

```diff
@@ -142,33 +172,40 @@ override 'execute' => sub {
     return 1;
 };
 
+has 'output_file' => (
+    is => 'ro',
+    isa => 'Str',
+    lazy_build => 1,
+);
+
+sub _build_output_file {
+    my $self = shift;
+    return catfile($self->qc_out->[0], $self->filename_root.'_gatk_collecthsmetrics.txt');
+}
+
 sub _parse_metrics {
     my ($self, $fh) = @_;
     if (!$fh) {
         croak q[File handle is not available, cannot parse picard pulldown metrics];
     }
     my @lines = ();
-    while (my $line = <$fh>) {
-        ## no critic (RequireExtendedFormatting)
-        if ( $line =~ /^#/sm ) { next; }
-        ## use critic
-        chomp $line;
-        if ($line =~ /^\s*$/smx) { next; }
-        push @lines, $line;
+    my $line = q{};
+    ## no critic (RequireExtendedFormatting)
+    while ($line !~ /^## METRICS/sm) {
+        $line = <$fh>;
     }
-    close $fh or croak qq[Cannot close pipe in __PACKAGE__ : $ERRNO, $CHILD_ERROR];
+    ## use critic
+    $lines[0] = <$fh>;
```

Comment:
This and next line - a test is needed that the value is defined. Error to say that the file is truncated if not.